panic converting untyped constant to uint

The following program `sample.go` triggers a panic:

```go
package main
func one() uint {
  return 1
}
func main() {
  println(one())
}
```

Expected result:
```console
$ go run ./sample.go
1
```

Got:
```console
$ yaegi ./sample.go
./sample.go:4:3: panic
./sample.go:9:11: panic
panic: reflect.Set: value of type int is not assignable to type uint [recovered]
	panic: reflect.Set: value of type int is not assignable to type uint [recovered]
	panic: reflect.Set: value of type int is not assignable to type uint

goroutine 1 [running]:
github.com/containous/yaegi/interp.runCfg.func1(0xc0002640a0, 0xc000185700)
	/home/max/go/src/github.com/containous/yaegi/interp/run.go:112 +0x218
panic(0xc99a00, 0xc0001704b0)
	/usr/local/go-1.13.3/src/runtime/panic.go:679 +0x1b2
github.com/containous/yaegi/interp.runCfg.func1(0xc000264140, 0xc000184c00)
	/home/max/go/src/github.com/containous/yaegi/interp/run.go:112 +0x218
panic(0xc99a00, 0xc0001704b0)
	/usr/local/go-1.13.3/src/runtime/panic.go:679 +0x1b2
reflect.Value.assignTo(0xc98900, 0xc0001720f0, 0x82, 0xdd7d72, 0xb, 0xc99c00, 0x0, 0xc000264140, 0xa0, 0xa0)
	/usr/local/go-1.13.3/src/reflect/value.go:2403 +0x432
reflect.Value.Set(0xc99c00, 0xc0001720f8, 0x187, 0xc98900, 0xc0001720f0, 0x82)
	/usr/local/go-1.13.3/src/reflect/value.go:1537 +0xbd
github.com/containous/yaegi/interp._return.func3(0xc000264140, 0x4a9cc1)
	/home/max/go/src/github.com/containous/yaegi/interp/run.go:1338 +0x8a
github.com/containous/yaegi/interp.runCfg(0xc000184c00, 0xc000264140)
	/home/max/go/src/github.com/containous/yaegi/interp/run.go:118 +0x7a
github.com/containous/yaegi/interp.call.func4(0xc0002640a0, 0xc99c00)
	/home/max/go/src/github.com/containous/yaegi/interp/run.go:758 +0xade
github.com/containous/yaegi/interp.runCfg(0xc000185700, 0xc0002640a0)
	/home/max/go/src/github.com/containous/yaegi/interp/run.go:118 +0x7a
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0002571e0, 0xc000184e00, 0xc000264000)
	/home/max/go/src/github.com/containous/yaegi/interp/run.go:96 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0002571e0, 0xc00026c000, 0x55, 0x255, 0xc00026c000, 0x55, 0x0, 0x1)
	/home/max/go/src/github.com/containous/yaegi/interp/interp.go:360 +0x42d
main.main()
	/home/max/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42d
```

If I define `func one() int { return 1 }` then it works.

--- a/_test/ret7.go
+++ b/_test/ret7.go
@@ -1,0 +1,11 @@
+package main
+
+func one() uint {
+	return 1
+}
+func main() {
+	println(one())
+}
+
+// Output:
+// 1

--- a/interp/run.go
+++ b/interp/run.go
@@ -1353,7 +1353,11 @@ func _return(n *node) {
 		case interfaceT:
 			values[i] = genValueInterface(c)
 		default:
-			values[i] = genValue(c)
+			if c.typ.untyped {
+				values[i] = genValueAs(c, def.typ.ret[i].rtype)
+			} else {
+				values[i] = genValue(c)
+			}
 		}
 	}
 

--- a/interp/run.go
+++ b/interp/run.go
@@ -1354,7 +1354,7 @@ func _return(n *node) {
 			values[i] = genValueInterface(c)
 		default:
 			if c.typ.untyped {
-				values[i] = genValueAs(c, def.typ.ret[i].rtype)
+				values[i] = genValueAs(c, def.typ.ret[i].TypeOf())
 			} else {
 				values[i] = genValue(c)
 			}

